crates/currency/src/lib.rs

```diff
@@ -0,0 +1,326 @@
+//! # Currency Module
+//! Based on the [Collateral specification](https://interlay.gitlab.io/polkabtc-spec/spec/collateral.html).
+//! Based on the [Treasury specification](https://interlay.gitlab.io/polkabtc-spec/spec/treasury.html).
+
+#![deny(warnings)]
+#![cfg_attr(test, feature(proc_macro_hygiene))]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+use frame_support::{
+    dispatch::{DispatchError, DispatchResult},
+    ensure,
+    traits::{Currency, ExistenceRequirement, ReservableCurrency},
+};
+
+pub type BalanceOf<T, I = ()> =
+    <<T as Config<I>>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+pub use pallet::*;
+
+pub type Collateral = pallet::Instance1;
+pub type Treasury = pallet::Instance2;
+
+pub const BTC_DECIMALS: u8 = 8;
+pub const DOT_DECIMALS: u8 = 10;
+
+#[frame_support::pallet]
+pub mod pallet {
```

Comment:
The currency pallet should additionally provide a "token" struct that defines the name and the symbol for the currency we are dealing with. That allows us to keep the JS code abstract as well and just define the symbols in one place without additional mapping. For my thinking around these things, also see the WIP currency types: https://www.notion.so/interlay/Public-Draft-Currency-Types-83c31bffdbc44316a3d4131ddc7c0089